# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/PerformanceStorageWorker.cpp
# Commit: 4f2ec6bc75c1
# Full Hash: 4f2ec6bc75c142b1633a8ae2a98b70fd5215ed1f
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-27 09:40:54
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Saving NavigationPreload's PerformanceTimingData into worker's PerformanceStorage. r=dom-worker-reviewers,jesup
#   
#   Depends on D138251
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138252
# ==============================================================================

diff -r 8fec60d0c499 -r 4f2ec6bc75c1 dom/performance/PerformanceStorageWorker.cpp
--- a/dom/performance/PerformanceStorageWorker.cpp	Sun Feb 27 00:26:22 2022 +0000
+++ b/dom/performance/PerformanceStorageWorker.cpp	Sun Feb 27 00:26:22 2022 +0000
@@ -120,6 +120,20 @@
   Unused << NS_WARN_IF(!r->Dispatch());
 }
 
+void PerformanceStorageWorker::AddEntry(
+    const nsString& aEntryName, const nsString& aInitiatorType,
+    UniquePtr<PerformanceTimingData>&& aData) {
+  MOZ_ASSERT(!NS_IsMainThread());
+  if (!aData) {
+    return;
+  }
+
+  UniquePtr<PerformanceProxyData> data = MakeUnique<PerformanceProxyData>(
+      std::move(aData), aInitiatorType, aEntryName);
+
+  AddEntryOnWorker(std::move(data));
+}
+
 void PerformanceStorageWorker::ShutdownOnWorker() {
   MutexAutoLock lock(mMutex);
 